### DISCUSSION TITLE
Request for Guidance on Applying Pressure-Time History to Structural Analysis in Kratos

### DISCUSSION THREAD

**Discussion by engrjessie** [KratosMultiphysics/Kratos#12664] [Q&A]
Hello everyone,

I’m new to Kratos and currently trying to use it for my research. I have successfully compiled Kratos on my machine and integrated it with GiD. However, I’m having difficulty understanding how to proceed with applying a pressure-time history (obtained from OpenFOAM simulations) to a structural analysis, particularly for masonry infills. I really wanted to try new things for structural analysis that is why I'm trying my best to learn Kratos and use it for my research.

My objective is to evaluate the structural response of these infills under the given pressure-time history using Kratos. I would greatly appreciate any guidance or suggestions on how to start. Can I do it in the Kratos-GiD module? Or if not, how to use the compiled version of Kratos on my machine.

Specifically, I am looking for help with:
- Importing and applying the pressure-time data from OpenFOAM.
- Setting up the structural model and boundary conditions for masonry infills (in Kratos or GiD).
- Any useful documentation, tutorials, or examples that could help me get started.

Thank you in advance for your help!

**Comment by loumalouomega** [comment]
Greetings, there is CoSimulation integration between Kratos and OpenFOAM. Take a look at this:

https://github.com/SADPR/CoSimIO-SolversCoupling

Thanks to @SADPR ! ❤️

**Comment by rubenzorrilla** [comment]
Hi @engrjessie. Did @loumalouomega 's suggestion work?

**Reply by engrjessie** [reply]
Hello @rubenzorrilla and @loumalouomega . Thank you so much for this, I am sorry for the late reply since I lost along the way. May I ask Dr. @loumalouomega is cosimulation installation could be done even after I have compiled kratos in my machine? Thank you in advance.

**Reply by loumalouomega** [reply]
You need to compile the cosimulation application as well (and the mappingapplication), these two apps are not big, so it would take no much time to compile.

**Reply by engrjessie** [reply]
Thank you for this information, @loumalouomega . I'll try to compile it together with the mapping application. I hope you'll still be here to cater to my future questions and clarifications.
